src/backend/src/controllers/change-requests.controllers.ts

```diff
@@ -22,7 +29,14 @@ export const getChangeRequestByID = async (req: Request, res: Response) => {
   if (requestedCR === null) {
     return res.status(404).json({ message: `change request with id ${crId} not found!` });
   }
-  return res.status(200).json(changeRequestTransformer(requestedCR));
+  try {
+    return res.status(200).json(changeRequestTransformer(requestedCR));
+  } catch (e) {
+    if (e instanceof TypeError) {
```

Comment:
I'm not so sure about the usage of TypeError. If you look up the docs it doesn't quite fit. If we think back to Java from fundies/OOD What we're missing is the different kinds of errors like illegal state, etc. Technically I think the ideal way is we subclass the Error object and make our own or something like that